### PR TITLE
PLT-2951 Improved signup process

### PR DESCRIPTION
**Pull request by DavidLu1997**
#### Summary
Added a page with links to each signup option
- Moved email and ldap out into their own pages
- Added links for all other signup options

UI vetted in https://github.com/mattermost/platform/pull/3704

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-2951

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization files updated

### CODE REVIEW

**Comment by esethna** [comment]
Reviewed. David is making a few updates regarding bugs in the LDAP "Login" button, GitLab username creation, and Office 365 login redirect.

**Comment by philku79** [comment]
There is an issue showing the pageindicators over the app symbols on 2nd page of the wizard.

**Comment by DavidLu1997** [comment]
@esethna I've spoken with @jwilander:
- Office365 is sort of broken with `bad token` errors when personal accounts are used, a comprehensive fix is being prepared (not related to this PR)
- LDAP can be tested on CI (as spinmints are http only), confirmed by JW
- I am currently working on GitLab configuration, but that can be tested on CI (with pre-configured, known-working tokens) if needed

**Comment by esethna** [comment]
Thanks @DavidLu1997, this looks good given the notes you've made above^. After dev review we can get this on master and test LDAP, Office (after fix goes in). I'm also Qing a discussion for UX on the button width and alignment, seems a bit random, but Asaad can submit a fix for that later on if we decide to make any changes.

File: webapp/components/signup/components/signup_email.jsx [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+        let email = '';
+        let teamDisplayName = '';
+        let teamName = '';
+        let teamId = '';
+        let loading = true;
+        let serverError = '';
+        let noOpenServerError = false;
+
+        if ((inviteId && inviteId.length > 0) || (hash && hash.length > 0)) {
+            if (hash) {
```
Comment (jwilander) [inline webapp/components/signup/components/signup_email.jsx]: This is a weird looking set of if statements, why not:
```
if (hash && hash.length > 0) {
    // do the hash stuff
} else if (inviteId && inviteId.length > 0) {
    // do the invite id stuff
} else {
    // do the else stuff
}
```

File: webapp/components/signup/components/signup_email.jsx [hunk 2]
```diff
@@ -0,0 +1,509 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
```
Comment (jwilander) [inline webapp/components/signup/components/signup_email.jsx]: New files should have the copyright year as 2016, can you update this in all new files?

File: webapp/components/signup/signup_controller.jsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        let usedBefore = false;
+
+        if (window.location.query) {
+            const hash = window.location.query.h;
+            const data = window.location.query.d;
+            const inviteId = window.location.query.id;
+
+            if ((inviteId && inviteId.length > 0) || (hash && hash.length > 0)) {
+                if (UserStore.getCurrentUser()) {
+                    Client.addUserToTeamFromInvite(
```
Comment (jwilander) [inline webapp/components/signup/signup_controller.jsx]: We shouldn't make HTTP requests inside the constructor, can you move this to `componentDidMount`? Also this if statement is a bit confusing, can you try to clean it up a bit?

**Comment by jwilander** [comment]
+1

**Comment by enahum** [comment]
+1

**Pull request merged**